### PR TITLE
Improve documentation, by clarifying certain parts and rephrasing others

### PR DESCRIPTION
**Pull request by louis-z**

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -27,7 +27,7 @@
 |[MA0024](https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0024.md)|Usage|Use an explicit StringComparer when possible|<span title='Warning'>⚠️</span>|✔️|✔️|
 |[MA0025](https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0025.md)|Design|Implement the functionality instead of throwing NotImplementedException|<span title='Warning'>⚠️</span>|✔️|❌|
 |[MA0026](https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0026.md)|Design|Fix TODO comment|<span title='Warning'>⚠️</span>|✔️|❌|
-|[MA0027](https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0027.md)|Usage|Do not remove original exception|<span title='Warning'>⚠️</span>|✔️|✔️|
+|[MA0027](https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0027.md)|Usage|Prefer rethrowing an exception implicitly|<span title='Warning'>⚠️</span>|✔️|✔️|
```
Comment (meziantou) [inline docs/README.md:30]: Do not edit the title directly. Instead, you must edit the code and run the DocumentationGenerator project

File: docs/README.md [hunk 2]
```diff
@@ -157,12 +157,12 @@ dotnet_diagnostic.MA0036.severity = suggestion # MA0036: Make class static
 dotnet_diagnostic.MA0037.severity = error      # MA0037: Remove empty statement
 dotnet_diagnostic.MA0038.severity = suggestion # MA0038: Make method static
 dotnet_diagnostic.MA0039.severity = error      # MA0039: Do not write your own certificate validation method
-dotnet_diagnostic.MA0040.severity = suggestion # MA0040: Flow the cancellation token
+dotnet_diagnostic.MA0040.severity = suggestion # MA0040: Pass along the cancellation token
```
Comment (meziantou) [inline docs/README.md]: I think you should copy the wording of https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2016

File: docs/Rules/MA0102.md [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 A struct member should use the `readonly` modifier when it does not modify state.
 
-note: This rule is enabled when using C# 8 or later
+> Note: This rule is enabled when using C# 8 or later
```
Comment (meziantou) [inline docs/Rules/MA0102.md]: Can you try the syntax explain in https://github.com/community/community/discussions/16925 to format note section?

**Review by meziantou — CHANGES_REQUESTED** [review]

File: docs/README.md (hunk 1 above)
Comment (louis-z) [reply docs/README.md:30]: Oh yeah, I had forgotten about that... My bad. Fixed it.

File: docs/Rules/MA0102.md (hunk 3 above)
Comment (louis-z) [reply docs/Rules/MA0102.md]: Seems to work. See [here](https://github.com/louis-z/Meziantou.Analyzer/blob/improve-doc/docs/Rules/MA0102.md).

File: src/Meziantou.Analyzer/Rules/MethodOverridesShouldNotChangeParameterDefaultsAnalyzer.cs [hunk 4]
```diff
@@ -10,8 +10,8 @@ public sealed class MethodOverridesShouldNotChangeParameterDefaultsAnalyzer : Di
 {
     private static readonly DiagnosticDescriptor s_rule = new(
         RuleIdentifiers.MethodOverridesShouldNotChangeParameterDefaults,
-        title: "Method overrides should not change parameter defaults",
-        messageFormat: "Method overrides should not change parameter defaults (original: {0}; current: {1})",
+        title: "Method overrides should not change default arguments",
```
Comment (meziantou) [inline src/Meziantou.Analyzer/Rules/MethodOverridesShouldNotChangeParameterDefaultsAnalyzer.cs]: I think we should use "default values" as in the documentation https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/named-and-optional-arguments?WT.mc_id=DT-MVP-5003978#optional-arguments

File: src/Meziantou.Analyzer/Rules/DoNotRemoveOriginalExceptionFromThrowStatementAnalyzer.cs [hunk 5]
```diff
@@ -11,8 +11,8 @@ public sealed class DoNotRemoveOriginalExceptionFromThrowStatementAnalyzer : Dia
 {
     private static readonly DiagnosticDescriptor s_rule = new(
         RuleIdentifiers.DoNotRemoveOriginalExceptionFromThrowStatement,
-        title: "Do not remove original exception",
-        messageFormat: "Do not remove original exception",
+        title: "Prefer rethrowing an exception implicitly",
```
Comment (meziantou) [inline src/Meziantou.Analyzer/Rules/DoNotRemoveOriginalExceptionFromThrowStatementAnalyzer.cs:14]: I'm not sure to understand why you say "implicitly". The rule force you to keep the original exception when throwing a new exception (exception wrapping). If you don't do it, there is no "implicit" exception, the original exception is lost.

Comment (louis-z) [reply src/Meziantou.Analyzer/Rules/DoNotRemoveOriginalExceptionFromThrowStatementAnalyzer.cs:14]: Isn't this rule equivalent to `CA2200: Rethrow to preserve stack details`? I based the description here on [that rule's example](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2200#example). The throw itself is not implicit, but the exception should be.

File: src/Meziantou.Analyzer/Rules/MethodOverridesShouldNotChangeParameterDefaultsAnalyzer.cs (hunk 4 above)
Comment (louis-z) [reply src/Meziantou.Analyzer/Rules/MethodOverridesShouldNotChangeParameterDefaultsAnalyzer.cs]: Done

**Review by meziantou — APPROVED** [review]

**Pull request merged**